### ISSUE TITLE
Missing indication EFO map for the drug Aspirin

### ISSUE THREAD

**Issue by mkarmona** [opentargets/platform#1217]
Having the chembl drug id `CHEMBL25` one of its indications is
```
              {
                "maxPhaseForIndication" : 4,
                "references" : [
                  {
                    "source" : "DailyMed",
                    "ids" : [
                      "03cdc2f8-78d1-11dc-8314-0800200c9a66"
                    ],
                    "urls" : [
                      "http://dailymed.nlm.nih.gov/dailymed/drugInfo.cfm?setid=03cdc2f8-78d1-11dc-8314-0800200c9a66"
                    ]
                  }
                ]
              },
```
@JarrodBaker and this indication does not contain EFO mapping so either does not include. @d0choa should we report to Chembl?

**Comment by mkarmona** [comment]
this is for the drug beta index

**Comment by JarrodBaker** [comment]
The problem seems to be coming from the raw ChEMBL data itself. Assume that we have loaded the raw chembl data into a DataFrame `inds`, then select the fields of interest: 

```
{
                 val aspirin = inds
                   .filter($"molecule_chembl_id" === "CHEMBL25")
                   .withColumn("refId", explode($"indication_refs.ref_id").as("refId"))
                   .withColumn("refType", explode($"indication_refs.ref_type").as("refType"))
                   .drop("indication_refs")
                   .filter($"refId" === "03cdc2f8-78d1-11dc-8314-0800200c9a66" && $"refType" === "DailyMed")
} 
```
We get the following output: 
```
+-----------+-----------------+------------------+--------------------+--------+
|     efo_id|max_phase_for_ind|molecule_chembl_id|               refId| refType|
+-----------+-----------------+------------------+--------------------+--------+
|       null|                4|          CHEMBL25|03cdc2f8-78d1-11d...|DailyMed|
|EFO:0003843|                4|          CHEMBL25|03cdc2f8-78d1-11d...|DailyMed|
|EFO:0003843|                4|          CHEMBL25|03cdc2f8-78d1-11d...|DailyMed|
|EFO:0003843|                4|          CHEMBL25|03cdc2f8-78d1-11d...|DailyMed|
|EFO:0003843|                4|          CHEMBL25|03cdc2f8-78d1-11d...|DailyMed|
|EFO:0003843|                4|          CHEMBL25|03cdc2f8-78d1-11d...|DailyMed|
|EFO:0003843|                4|          CHEMBL25|03cdc2f8-78d1-11d...|DailyMed|
|EFO:0003843|                4|          CHEMBL25|03cdc2f8-78d1-11d...|DailyMed|
+-----------+-----------------+------------------+--------------------+--------+
```
The [data-pipeline](https://github.com/opentargets/data_pipeline/blob/469bb6ca51381456b60ba6724f65f4d2e764bb9b/mrtarget/modules/Drug.py#L317) took the approach of not including any indications which did not have an associated `efo_id`. 

If there are no objections, I'll take the same approach and throw away any indications that don't have an `efo_id`.

**Comment by d0choa** [comment]
@JarrodBaker this sounds good to me. I would suggest to include another field named `indicationFromSource` which contains the indication label from ChEMBL. This might help us identify if something fishy is going on with the indication mapping.

Since 1,400 missed indications is a significant number, @JarrodBaker could you pass to @ireneisdoomed a table with all the indication labels - ChEMBL ids pairs for the unmapped ? She can try to run it through Zooma/Ontoma and check if there are some obvious mappings that ChEMBL could rescue.

If this number is still low, we might have to list indications as unresolved entities using the `indicationFromSource` field.

**Comment by d0choa** [comment]
For completeness, the above listed unmapped indication corresponds to `Spasm`, which in my opinion should have been mapped to the EFO term [HP_0003394  - `muscle spasm`](http://purl.obolibrary.org/obo/HP_0003394) ([OT platform alpha](https://alpha.targetvalidation.org/disease/HP_0003394))

![Screenshot 2020-10-20 at 13 22 09](https://user-images.githubusercontent.com/5097586/96585340-5f3c7b80-12d7-11eb-84fe-5feaf71c451a.png)

**Comment by JarrodBaker** [comment]
@d0choa I've pulled the indication labels from the ChEMBL ElasticSearch (I'm guessing you're referring to what they call the `efo_term`). In all cases where the `efo_id` is null the `efo_term` is also `null`. 

I've created a csv file in the following form:
```
root
 |-- molecule_chembl_id: string (nullable = true)
 |-- efo_id: string (nullable = true)
 |-- efo_term: string (nullable = true)
 |-- refId: string (nullable = true)
 |-- refType: string (nullable = true)
 |-- refUrl: string (nullable = true)
```
and uploaded it to `gs://ot-team/drop/indicationsWithNulls.csv` for @ireneisdoomed to see if she can make any headway. 

How did you find the original term? Indications that were generated using the old pipeline which had no `efo_id` were being dropped?

Given there are no labels coming out of ChEMBL, shall I just save the references for the time being in something like `indicationsWithoutEfo`?

**Comment by d0choa** [comment]
I suspect the above label `Spasm` would be somewhere in their ES index for aspirin. That's the field, I was referring to. It's the raw indication, before any mapping to EFO is performed.

**Comment by ireneisdoomed** [comment]
Having used the `mesh_heading` Jarrod has fetched from the their ES, OnToma is able to map nearly 60% of the unmapped MeSH terms. You can find the table here `gs://ot-team/drop/matched_indications.csv`. For the case given, `Spasm` has been mapped to [muscle cramp](https://www.ebi.ac.uk/ols/ontologies/efo/terms?short_form=EFO_0009846).

Among the unsuccessful results, these 6 terms do not map to any EFO: `'Electrolytes', 'Peritonitis', 'Vaginismus', 'Chancroid', 'Diverticulum, Colon', 'Cholestasis'`:

- Electrolytes: No match, too unespecific
- Peritonitis: EFO entry, OnToma does not find it? (_EFO_0008588_)
- Vaginismus: not in EFO but MONDO ID available: _MONDO_0021723_
- Chancroid: not in EFO but MONDO ID available: _MONDO_0001797_
- Diverticulum, Colon: No match
- Cholestasis: EFO entry including MONDO, OnToma does not find it? (MONDO_0001751) 

An error has been identified in the OnToma tool as both peritonitis and cholestasis should be mapped to an exact match. This is addressed in #1220.

Regarding `Vaginismus` and `Chancroid`, should we inform EFO to include these?

**Comment by JarrodBaker** [comment]
@mkarmona and I discussed this bug this morning at our regular backend meeting. The decision for the time being is to wait for ChEMBL to update their data with the missing values that @ireneisdoomed found. While it's technically feasible for us to manually patch in Irene's data, it adds another dependency to the ETL and makes it harder to understand the code.
